Code/EditorPlugins/Assets/EnginePluginAssets/RenderPipelineAsset/RenderPipelineContext.cpp

```diff
@@ -0,0 +1,146 @@
+
+ezRenderPipelineContext::ezRenderPipelineContext()
+  : ezEngineProcessDocumentContext(ezEngineProcessDocumentContextFlags::CreateWorld)
+{
+}
+
+void ezRenderPipelineContext::HandleMessage(const ezEditorEngineDocumentMsg* pMsg)
+{
+  ezEngineProcessDocumentContext::HandleMessage(pMsg);
+}
+
+void ezRenderPipelineContext::OnInitialize()
+{
+}
+
+ezEngineProcessViewContext* ezRenderPipelineContext::CreateViewContext()
+{
+  EZ_ASSERT_DEV(false, "Should not be called");
+  return nullptr;
+}
+
+void ezRenderPipelineContext::DestroyViewContext(ezEngineProcessViewContext* pContext)
+{
+  EZ_ASSERT_DEV(false, "Should not be called");
+}
+
+ezWorldRttiConverterContext& ezRenderPipelineContext::GetContext()
+{
+  return m_RenderPipelineContext;
+}
+
+const ezWorldRttiConverterContext& ezRenderPipelineContext::GetContext() const
+{
+  return m_RenderPipelineContext;
+}
+
+ezStatus ezRenderPipelineContext::ExportDocument(const ezExportDocumentMsgToEngine* pMsg)
+{
+  ezDynamicArray<ezRenderPipelinePass*> passes;
+  ezDynamicArray<ezExtractor*> extractors;
+  ezDynamicArray<ezRenderPipelineResourceLoaderConnection> connections;
+
+  ezDynamicArray<ezUuid> passUuids;
+  ezDynamicArray<ezRenderPipelineContextLoaderConnection*> toolConnections;
+
+  m_RenderPipelineContext.GetObjectsByType(passes, &passUuids);
+  m_RenderPipelineContext.GetObjectsByType(extractors);
+  m_RenderPipelineContext.GetObjectsByType(toolConnections);
+
+  ezHashTable<ezUuid, ezUInt32> passUuidToIndex;
+  for (ezUInt32 i = 0; i < passUuids.GetCount(); ++i)
+  {
+    passUuidToIndex.Insert(passUuids[i], i);
+  }
+  connections.SetCount(toolConnections.GetCount());
+  for (ezUInt32 i = 0; i < toolConnections.GetCount(); i++)
+  {
+    ezRenderPipelineContextLoaderConnection* pConnection = toolConnections[i];
+    ezRenderPipelineResourceLoaderConnection& engineConnection = connections[i];
+    EZ_VERIFY(passUuidToIndex.TryGetValue(pConnection->m_Source, engineConnection.m_uiSource), "");
+    EZ_VERIFY(passUuidToIndex.TryGetValue(pConnection->m_Target, engineConnection.m_uiTarget), "");
+    engineConnection.m_sSourcePin = pConnection->m_SourcePin;
+    engineConnection.m_sTargetPin = pConnection->m_TargetPin;
+  }
+
+  ezDefaultMemoryStreamStorage storage;
+  {
+    // Export Resource Data
+    ezMemoryStreamWriter writer(&storage);
+    EZ_SUCCEED_OR_RETURN(ezRenderPipelineResourceLoader::ExportPipeline(passes.GetArrayPtr(), extractors.GetArrayPtr(), connections.GetArrayPtr(), writer));
+  }
+
+  ezDeferredFileWriter file;
+  file.SetOutput(pMsg->m_sOutputFile);
+
+  {
+    // File Header
+    ezAssetFileHeader header;
+    header.SetFileHashAndVersion(pMsg->m_uiAssetHash, pMsg->m_uiVersion);
+    header.Write(file).IgnoreResult();
+
+    ezUInt8 uiVersion = 2;
+    file << uiVersion;
+  }
+
+  {
+    // Resource Data
+    ezUInt32 uiSize = storage.GetStorageSize32();
+    file << uiSize;
+    storage.CopyToStream(file).AssertSuccess("Failed to copy to file writer");
```

Comment:
@Sanakan8472 any reason why this is an `AssertSuccess` and not a `return ezStatus("io failed")` ?

Reply:
Fixed, thanks!